bofire/data_models/acquisition_functions/acquisition_function.py

```diff
@@ -17,55 +18,59 @@ class MultiObjectiveAcquisitionFunction(AcquisitionFunction):
     type: str
 
 
-class qNEI(SingleObjectiveAcquisitionFunction):
+class MCAcquisitionFunction(BaseModel):
+    n_mc_samples: TIntPowerOfTwo = 512
+
+
+class qNEI(SingleObjectiveAcquisitionFunction, MCAcquisitionFunction):
     type: Literal["qNEI"] = "qNEI"
     prune_baseline: bool = True
 
 
-class qLogNEI(SingleObjectiveAcquisitionFunction):
+class qLogNEI(SingleObjectiveAcquisitionFunction, MCAcquisitionFunction):
     type: Literal["qLogNEI"] = "qLogNEI"
     prune_baseline: bool = True
 
 
-class qEI(SingleObjectiveAcquisitionFunction):
+class qEI(SingleObjectiveAcquisitionFunction, MCAcquisitionFunction):
     type: Literal["qEI"] = "qEI"
 
 
-class qLogEI(SingleObjectiveAcquisitionFunction):
+class qLogEI(SingleObjectiveAcquisitionFunction, MCAcquisitionFunction):
     type: Literal["qLogEI"] = "qLogEI"
 
 
-class qSR(SingleObjectiveAcquisitionFunction):
+class qSR(SingleObjectiveAcquisitionFunction, MCAcquisitionFunction):
     type: Literal["qSR"] = "qSR"
 
 
-class qUCB(SingleObjectiveAcquisitionFunction):
+class qUCB(SingleObjectiveAcquisitionFunction, MCAcquisitionFunction):
     type: Literal["qUCB"] = "qUCB"
     beta: Annotated[float, Field(ge=0)] = 0.2
 
 
-class qPI(SingleObjectiveAcquisitionFunction):
+class qPI(SingleObjectiveAcquisitionFunction, MCAcquisitionFunction):
     type: Literal["qPI"] = "qPI"
     tau: PositiveFloat = 1e-3
 
 
-class qEHVI(MultiObjectiveAcquisitionFunction):
+class qEHVI(MultiObjectiveAcquisitionFunction, MCAcquisitionFunction):
```

Comment:
Why did you add a parameter via inheritance? I would just put the parameter there. As in the case of `alpha`. If you really want to gather the `MCAcquisitionFunction` parameters somehow because you expect to add more parameters to `MCAcquisitionFunction` in the near future, at least do it via aggregation.

Reply:
Hmm, why not adding the param via inheritance? And what do you mean by aggregation?

Reply:
I personally think we should the code as simple as possible. This way I think, it is hard to read. You look at the data model and want to see the parameters. But there are none. I very often have a hard time in to find the class that actually implements the functionality I am looking for in BoFire due to its deep inheritance hierarchy. Ok, readability is subjective. 

With aggregation I meant composition. I would only use inheritance if I want to make use of polymorphism but not just to save some characters. I consider it good practice in object oriented programming to favor composition over inheritance. https://en.wikipedia.org/wiki/Composition_over_inheritance

Reply:
Hmm, I prefer the inheritance :D 

But I will change it ;)

Best,

Johannes

Reply:
> Hmm, I prefer the inheritance :D

I know 😉